org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XSISchemaModel.java

```diff
@@ -0,0 +1,102 @@
+package org.eclipse.lsp4xml.services;
+
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionItemKind;
+import org.eclipse.lsp4j.CompletionList;
+import org.eclipse.lsp4j.InsertTextFormat;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4xml.dom.Element;
+import org.eclipse.lsp4xml.dom.XMLDocument;
+import org.eclipse.lsp4xml.services.extensions.ICompletionResponse;
+
+/**
+ * This class holds values that represent the XSI xsd.
+ * Can be seen at https://www.w3.org/2001/XMLSchema-instance
+ */
+public class XSISchemaModel {
+
+	public static final String XSI_NIL = "nil";
+	public static final String XSI_TYPE = "type";
+	public static final String XSI_SCHEMA = "schemaLocation";
+	public static final String XSI_NOSCHEMA = "noNamespaceSchemaLocation";
+	
+	/**
+	 * Checks if attribute name has a suffix (eg: attName:SUFFIX)
+	 * 
+	 * If attName ends with ':' it is considered to not have a suffix value
+	 * @param attName
+	 * @return attName has suffix, ending with ':' returns true
+	 */
+	public static boolean hasSuffixValue(String attName) {
+		int i;
+		for(i = 0; i < attName.length(); i++) {
+			char c = attName.charAt(i);
+			if(c == ':') {
+				break;
+			}
+		}
+		return i != attName.length() - 1;
+	}
+
+	public static String getPrefixFromAttName(String attName) {
+		int i;
+		for(i = 0; i < attName.length(); i++) {
+			char c = attName.charAt(i);
+			if(c == ':') {
+				break;
+			}
+		}
+		return i != attName.length() ? attName.substring(0, i) : attName;
+	}
+
+	private static boolean startMatchesAttribute(String actualPrefix, String suffixConstant, String attName) {
+		return (actualPrefix + ":" + suffixConstant).startsWith(attName);
+	}
+
+	public static void computeCompletionResponses(ICompletionResponse response, boolean isSnippetsSupported, String actualPrefix, String attName, Range editRange, XMLDocument document) {
+		String snippet = "";
+		if(isSnippetsSupported) {
+			snippet = "$0";
+		}
+		String name;
+		Element root = document.getDocumentElement();
+		if(!attributeAlreadyExists(root, actualPrefix, "nil") && startMatchesAttribute(actualPrefix, XSI_NIL, attName)) {
```

Comment:
I think we must not manage startMatchesAttribute, because you break th ecapability the fuzzy completion (ex: you could type "schema" and completion will provide schemaLocation and noNamespaceSchemaLocation.